### DISCUSSION TITLE
Create Native library in .Net

### DISCUSSION THREAD

**Discussion by laibulle** [dotnet/maui#639] [Q&A]
Hello, 

is there a way in MAUI/.net6 to create a native library for iOS and Android like go, rust, Kotlin, C++ and many other languages can do? 

The goal would be to write C# logic and use it inside ReactNative or Flutter.

https://github.com/golang/go/wiki/Mobile
https://mozilla.github.io/firefox-browser-architecture/experiments/2017-09-06-rust-on-ios.html
https://kotlinlang.org/docs/apple-framework.html

The only thing I found doesn't seems to be a robust or futur proof solution https://github.com/mono/Embeddinator-4000

Best regard

**Comment by laibulle** [comment]
![image](https://user-images.githubusercontent.com/2327626/113826880-f365a680-9782-11eb-9e34-bc00ccc9523e.png)

@davidortinau do you have any idea ?

**Reply by GeraudFabien** [reply]
What you ask is neither a bug or a feature so you may ask on stackoverflow. 
Since you want to have only the logic in .Net not the UI you're question has nothinks to do with maui. Maybe you should ask it in dotnet/Runtime.
I don't know the answer to you're question but it seems to be really specific. So i don't thinks you will have a lot of answer. I have read somewhere that you could use Wasm with reactNative and someone as try to use C# with reactNative using Wasm. But i dont know the state of the project.
I'm sorry i can't help you more.

**Reply by laibulle** [reply]
Thanks @GeraudFabien, 

using Was is not an option because ReactNative use a browser context and I need to open TCP connections.

I will try to ask in the runtime repo as you suggest.

Best regards
